llvm/lib/Target/AMDGPU/SIInstructions.td

```diff
@@ -3400,7 +3400,7 @@ def : GCNPat<
 // FIXME: Should also do this for readlane, but tablegen crashes on
 // the ignored src1.
 def : GCNPat<
-  (int_amdgcn_readfirstlane (i32 imm:$src)),
+  (i32 (AMDGPUreadfirstlane (i32 imm:$src))),
```

Comment:
We might need to make this fold more sophisticated for other types, but best for a follow up patch